pkg/admission/prep_downscale_test.go

```diff
@@ -258,6 +259,44 @@ func testPrepDownscaleWebhook(t *testing.T, oldReplicas, newReplicas int, option
 	}
 }
 
+func TestFindStatefulSetWithNonUpdatedReplicas(t *testing.T) {
+	namespace := "test"
+	rolloutGroup := "ingester"
+	labels := map[string]string{config.RolloutGroupLabelKey: rolloutGroup}
+	objects := []runtime.Object{
+		&apps.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "zone-a",
+				Namespace: namespace,
+				Labels:    labels,
+			},
+			Status: apps.StatefulSetStatus{
+				Replicas:        10,
+				ReadyReplicas:   10,
+				UpdatedReplicas: 10,
+			},
+		},
+		&apps.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "zone-b",
+				Namespace: namespace,
+				Labels:    labels,
+			},
+			Status: apps.StatefulSetStatus{
+				Replicas:        10,
+				ReadyReplicas:   4,
+				UpdatedReplicas: 6,
+			},
+		},
+	}
+	api := fake.NewSimpleClientset(objects...)
+
+	sts, err := findStatefulSetWithNonUpdatedReplicas(context.Background(), api, namespace, rolloutGroup)
+	assert.Nil(t, err)
```

Comment:
Probably want `require.Method()` for these since `assert` doesn't stop execution on failure (so you'll get a `nil` pointer dereference)